packages/venia-ui/lib/components/RichContent/PageBuilder/ContentTypes/Tabs/tabs.js

```diff
@@ -0,0 +1,297 @@
+            navigationWrapper.childNodes[0].nodeName === 'UL'
+        ) {
+            navScrollElement = navigationWrapper.childNodes[0];
+            setScrollElement(navScrollElement);
+            // If there are additional tabs hidden by scroll we display a gradient on the right
+            if (navScrollElement.scrollWidth > navScrollElement.offsetWidth) {
+                setNavWrapperClass(classes.navigationGradientRight);
+            }
+            navScrollElement.addEventListener('scroll', handleScroll);
+        }
+
+        return () => {
+            if (navScrollElement) {
+                navScrollElement.removeEventListener('scroll', handleScroll);
+            }
+        };
+    }, [
+        classes.navigationGradientBoth,
+        classes.navigationGradientLeft,
+        classes.navigationGradientRight,
+        navigationRef
+    ]);
+
+    if (!headers.length) {
+        return null;
+    }
+
+    const wrapperStyles = {
+        marginTop,
+        marginRight,
+        marginBottom,
+        marginLeft,
+        paddingTop,
+        paddingRight,
+        paddingBottom,
+        paddingLeft
+    };
+
+    const contentStyles = {
+        minHeight,
+        textAlign
+    };
+
+    const tabWrapperProps = {
+        defaultIndex
+    };
+
+    if (border) {
+        wrapperStyles['--tabs-border'] = border;
+        wrapperStyles['--tabs-border-color'] = borderColor;
+    }
+    if (borderWidth) {
+        wrapperStyles['--tabs-border-width'] = borderWidth;
+    }
+    if (borderRadius) {
+        wrapperStyles['--tabs-border-radius'] = borderRadius;
+    }
+
+    const tabPanels = Children.map(children, (child, index) => {
+        return (
+            <TabPanel
+                key={index}
+                className={classes.panel}
+                selectedClassName={classes.panelSelected}
+            >
+                {child}
+            </TabPanel>
+        );
+    });
+
+    const navigationClass =
+        classes[
+            `navigation${tabNavigationAlignment.charAt(0).toUpperCase() +
+                tabNavigationAlignment.slice(1)}`
+        ];
+
+    const contentClass =
+        classes[
+            `content${tabNavigationAlignment.charAt(0).toUpperCase() +
+                tabNavigationAlignment.slice(1)}`
+        ];
+
+    return (
+        <TabWrapper
+            style={wrapperStyles}
+            className={[classes.root, ...cssClasses].join(' ')}
+            disabledTabClassName={classes.disabled}
+            selectedTabClassName={classes.selected}
+            {...tabWrapperProps}
+        >
+            <div className={navWrapperClass} ref={navigationRef}>
+                <TabList
+                    onMouseDown={onMouseDown}
+                    onMouseUp={onMouseUp}
+                    onMouseMove={onMouseMove}
+                    onMouseLeave={onMouseUp}
+                    className={navigationClass}
+                >
+                    {headers.map((header, i) => (
```

Comment:
Please avoid multiline expressions such as this in JSX, especially mapping. The way you defined `tabPanels` is ideal. 👍

Reply:
Look at us, being inconsistent. Resolved: https://github.com/magento/pwa-studio/pull/1872/commits/dc8fb48bc7c2edba5bd3573da6e9fa28e48adfc1